### PR TITLE
Fix license check for presubmit tests.

### PR DESCRIPTION
**Pull request by markusthoemmes**
While licenses are indeed checked during the test run, and an error is printed, the test does not fail as it should. This fixes that condition.

### CODE REVIEW

**Comment by markusthoemmes** [comment]
This confirms the check now correctly fails with:

```
Error identifying licenses for transitive dependencies: unable to find license for "github.com/knative/pkg/vendor/github.com/rogpeppe/go-internal/modfile"
```

File: hack/update-deps.sh [hunk 1]
```diff
@@ -25,6 +25,8 @@ cd ${REPO_ROOT_DIR}
 # Ensure we have everything we need under vendor/
 dep ensure
 
+check_licenses $(go list ./...)
```
Comment (mattmoor) [inline hack/update-deps.sh]: I don't think this is particularly fast (not that `dep` is), and we already run it in presubmit tests...

Comment (markusthoemmes) [reply hack/update-deps.sh]: Took it out of here for the presubmit test to catch.

**Pull request merged**